hoot-core-test/src/test/cpp/hoot/core/ops/UnconnectedWaySnapperTest.cpp

```diff
@@ -156,6 +158,35 @@ class UnconnectedWaySnapperTest : public HootTestFixture
     uut.setWayNodeToSnapToCriterionClassName(" ");
     HOOT_STR_EQUALS("hoot::WayNodeCriterion", uut._wayNodeToSnapToCriterionClassName);
   }
+
+  void runStaticSnapTest()
+  {
+    const QString testName = "runStaticSnapTest";
+
+    OsmXmlReader reader;
+    OsmMapPtr map(new OsmMap());
+    reader.setUseDataSourceIds(true);
+    reader.setDefaultStatus(Status::Unknown1);
+    reader.read(inputPath + testName + "In1.osm", map);
+    reader.setDefaultStatus(Status::Unknown2);
+    reader.read(inputPath + testName + "In2.osm", map);
+
+    MapProjector::projectToPlanar(map);
+
+    WayPtr way1 = map->getWay(FindWaysVisitor::findWaysByTag(map, "note", "way1")[0]);
+    WayPtr way2 = map->getWay(FindWaysVisitor::findWaysByTag(map, "note", "way2")[0]);
+    WayPtr way3 = map->getWay(FindWaysVisitor::findWaysByTag(map, "note", "way3")[0]);
+
+    UnconnectedWaySnapper::snapClosestEndpointToWay(map, way2, way1);
+    UnconnectedWaySnapper::snapClosestEndpointToWay(map, way3, way1);
+
+    MapProjector::projectToWgs84(map);
+    OsmXmlWriter writer;
+    writer.setIsDebugMap(true);
+    writer.write(map, outputPath + testName +  + "Out.osm");
+
+    HOOT_FILE_EQUALS(inputPath + testName +  + "Out.osm", outputPath + testName +  + "Out.osm");
+  }
```

Comment:
This test loads two files, one with a couple of roads that are very close to the roundabout in the other file.  The snapper snaps those endpoints to the roundabout so nothing is disconnected.